src/typecheck.rs

```diff
@@ -397,68 +225,43 @@ pub fn typeck_function(fun: &Function, ctx: &TypeCtx) -> Result<(), TypeError> {
 
     let e = typeck_expr(&fun.body, &fun_ctx)?;
 
-    //check that body is the same type as the return type, otherwise return error
+    // check that body is the same type as the return type, otherwise return error
+    // todo make sure to log the error
     if e == fun.ret_typ {
         Ok(())
     } else {
         let t = fun.ret_typ.clone();
-        Err(TypeError::TypeMismatch {
-            expected: t,
-            actual: e,
-        })
+        debug!("In: typeck_function {fun:?}");
+        error!("Body of {fun:?} not same type as return type. Body : {e:?}, Return Type: {t:?}");
```

Comment:
Don't log `error!` unless there is an error/inconsistency **in the program you're writing** and you're still continuing execution (which should ***never*** happen in the typechecker). You're already returning an `Err` for the type errors **in the uncrustable program** (which is the **correct behavior for your typechecker**).

Reply:
also fix this in all other places you use `error!`